src/handlers.cpp

```diff
@@ -1713,6 +1743,7 @@ void RegistrationTerminationTask::get_assoc_primary_public_ids_failure(Cassandra
   SAS::Event event(this->trail(), SASEvent::DEREG_SUCCESS, 0);
```

Comment:
I think this should be DEREG_FAIL instead of DEREG_SUCCESS - do you agree? (I realise this isn't your code.)